src/overlays/actors/ovl_En_Okuta/z_en_okuta.c

```diff
@@ -40,153 +55,1086 @@ ActorInit En_Okuta_InitVars = {
+    Animation_PlayOnce(&this->skelAnime, &gOctorokAppearAnim);
+    EnOkuta_SpawnBubbles(this, play);
+    this->actionFunc = EnOkuta_Appear;
+}
+
+/**
+ * Plays the appear animation to completion while scaling the Octorok to its full size, then transitions to the idle
+ * floating state. If the Octorok is a red Octorok, and the player is too close to the Octorok when its appear animation
+ * is complete, then the Octorok will instead hide underwater.
+ */
+void EnOkuta_Appear(EnOkuta* this, PlayState* play) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if ((this->actor.xzDistToPlayer < 160.0f) && (EN_OKUTA_GET_TYPE(&this->actor) == EN_OKUTA_TYPE_RED_OCTOROK)) {
+            EnOkuta_SetupHide(this);
+        } else {
+            EnOkuta_SetupFloat(this);
+        }
+    } else {
+        f32 curFrame = this->skelAnime.curFrame;
+
+        if (curFrame <= 4.0f) {
+            Actor_SetScale(&this->actor, curFrame * 0.25f * 0.01f);
+        } else if (Animation_OnFrame(&this->skelAnime, 5.0f)) {
+            Actor_SetScale(&this->actor, 0.01f);
+        }
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 2.0f)) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_OCTAROCK_JUMP);
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 12.0f)) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_DAIOCTA_LAND);
+    }
+
+    if ((Animation_OnFrame(&this->skelAnime, 3.0f)) || (Animation_OnFrame(&this->skelAnime, 15.0f))) {
+        EnOkuta_SpawnSplash(this, play);
+    }
+}
+
+void EnOkuta_SetupHide(EnOkuta* this) {
+    Animation_PlayOnce(&this->skelAnime, &gOctorokHideAnim);
+    this->actionFunc = EnOkuta_Hide;
+}
+
+/**
+ * Retreats underwater while scaling the Octorok down, then makes the Octorok wait to appear again.
+ */
+void EnOkuta_Hide(EnOkuta* this, PlayState* play) {
+    Math_ApproachF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.5f, 30.0f);
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_COMMON_WATER_MID);
+        EnOkuta_SpawnBubbles(this, play);
+        EnOkuta_SetupWaitToAppear(this);
+    } else {
+        f32 curFrame = this->skelAnime.curFrame;
+
+        if (curFrame >= 4.0f) {
+            Actor_SetScale(&this->actor, (6.0f - curFrame) * 0.5f * 0.01f);
+        }
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 2.0f)) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_DAIOCTA_SINK);
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 4.0f)) {
+        EnOkuta_SpawnSplash(this, play);
+    }
+}
+
+void EnOkuta_SetupFloat(EnOkuta* this) {
+    Animation_PlayLoop(&this->skelAnime, &gOctorokFloatAnim);
+
+    // `EnOkuta_Float` uses the `timer` variable in an unconventional way. It is used to track how many times the
+    // Octorok's float animation needs to loop before the Octorok can start shooting at the player. Note that the unused
+    // blue Octorok variant can ignore this variable and start shooting the player anyway if the player gets too close.
```

Comment:
```suggestion
    // `EnOkuta_Float` uses `timer` to store the number of times the float animation loops before firing at the player.
```

Can cut the fat here to get to the point quicker. Perhaps add a comment where `timer` is defined as well. 
Comments about Blue Octorok's behavior are better off left in `EnOkuta_Float` where the described behavior is actually implemented, since both types of Octoroks are using the value of `timer` in the same way.